pyccel/codegen/printing/ccode.py

```diff
@@ -645,6 +645,54 @@ def _handle_inline_func_call(self, expr):
 
         return code
 
+    def init_stc_container(self, expr, container_var):
+        """
+        Generate the initialization of an STC container in C.
+
+        This method generates and prints the C code for initializing a container using the STC `init()` method.
+
+        Parameters
+        ----------
+        expr : TypedAstNode
+            The object representing the container being printed (e.g., PythonList, PythonSet).
+    
+        container_var : str
+            The variable name to which the container is being assigned.
+        
+        Returns
+        -------
+        str
+            The generated C code for the container initialization.
+        """
+
+        dtype = self.find_in_type_registry(container_var.lhs.class_type)
+        keyraw = '{' + ', '.join(self._print(a) for a in expr.args) + '}'
+        init = f'{container_var.lhs.name} = c_init({dtype}, {keyraw});\n'
```

Comment:
See the [docs](https://github.com/pyccel/pyccel/blob/devel/developer_docs/codegen_stage.md):
> It can be tempting to skip some of these `_print` calls, especially for basic types such as `Literal`s.
> However it is very important to use these functions as much as possible, for several reasons:
> 1.  It ensures that the same conventions are used throughout the generated code
> 2.  It ensures that code details are not forgotten
> 3.  It makes refactoring much simpler
>
> For example, it is sometimes necessary to print a `0`.
> It is very tempting to just use `'0'` directly, however this would mean that the precision of the integer would not be printed.
> In some languages (e.g. Fortran) not specifying the precision in literals can have unintended consequences on the final result.

For example, your code will break with both of these examples:
```python
class A:
    def __init__(self, a):
        self.a = [1,2,3]
```

```python
def f():
    a = [1,2,3]
    b = [3,4,5]
    return a,b
```

Reply:
fixed 5c72144c6a07a1f8cd40885d15e1da677993abc0